Bottom: 64f7dc23bc7e514b3750762ee6eb4d111743c6f3
Top:    c1a5936161f8ccf9c2ed459cae5bbfd2c583e808
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-09-15 17:44:09 +0100

Refresh of report-c-don-t-leave-actions-i

---

diff --git a/src/report.c b/src/report.c
index 6801704..5cc58b9 100644
--- a/src/report.c
+++ b/src/report.c
@@ -556,6 +556,10 @@ XYtoNetLength (Coord x, Coord y, int *found)
 
   length = 0;
   *found = 0;
+
+  /* NB: The third argument here, 'false' ensures LookupConnection
+   *     does not add its changes to the undo system.
+   */
   LookupConnection (x, y, false, PCB->Grid, FOUNDFLAG);
 
   ALLLINE_LOOP (PCB->Data);
@@ -664,6 +668,7 @@ ReportAllNetLengths (int argc, char **argv, Coord x, Coord y)
 
           length = XYtoNetLength (x, y, &found);
 
+          /* Reset connectors for the next lookup */
           ResetConnections (false);
 
           pcb_sprintf(buf, "%$m*", units_name, length);
@@ -893,12 +898,12 @@ ReportNetLengthByName (char *tofind, int x, int y)
 #endif
 
   /* Reset all connection flags and save an undo-state to get back
-   * to the state the board was in when we started this function.
+   * to the state the board was in when we started.
    *
    * After this, we don't add any changes to the undo system, but
    * ensure we get back to a point where we can Undo() our changes
    * by resetting the connections with ResetConnections() before
-   * calling Undo() at the end of the procedure.
+   * calling Undo() when we are finished.
    */
   ResetConnections (false);
   IncrementUndoSerialNumber ();
@@ -906,6 +911,9 @@ ReportNetLengthByName (char *tofind, int x, int y)
   length = XYtoNetLength (x, y, &found);
   netname = net->Name + 2;
 
+  ResetConnections (false);
+  Undo (false);
+
   if (!found)
     {
       if (net_found)
@@ -913,7 +921,7 @@ ReportNetLengthByName (char *tofind, int x, int y)
       else
         gui->log ("Net not found.\n");
 
-      goto out;
+      return 1;
     }
 
   retval = 0;
@@ -927,10 +935,6 @@ ReportNetLengthByName (char *tofind, int x, int y)
       gui->log ("Net length: %s\n", buf);
   }
 
-out:
-  ResetConnections (false);
-  Undo (false);
-
   return retval;
 }
